src/roiextractors/multiimagingextractor.py

```diff
@@ -97,16 +106,53 @@ def get_frames(self, frame_idxs: ArrayType, channel: Optional[int] = 0) -> Numpy
     ) -> np.ndarray:
-        # To-do: implement this without reference to get_frames
+        if channel != 0:
+            raise NotImplementedError(
+                f"MultiImagingExtractors for multiple channels have not yet been implemented! (Received '{channel}'."
+            )
+
         start = start_frame if start_frame is not None else 0
         stop = end_frame if end_frame is not None else self.get_num_frames()
-        frame_idxs = range(start, stop)
-        return self.get_frames(frame_idxs=frame_idxs, channel=channel)
+        total_range = list(range(start, stop))
+        extractors_spanned = np.unique(np.searchsorted(self._end_frames, total_range, side="right"))
 
-    def _get_frames_from_an_imaging_extractor(self, extractor_index: int, frame_idxs: ArrayType) -> NumpyArray:
-        imaging_extractor = self._imaging_extractors[extractor_index]
-        frames = imaging_extractor.get_frames(frame_idxs=frame_idxs)
-        return frames
+        # Early return with simple relative indexing; preserves native return class of that extractor
+        if len(extractors_spanned) == 1:
+            relative_start = start - self._start_frames[extractors_spanned[0]]
+            relative_stop = stop - start + relative_start
+
+            return self._imaging_extractors[extractors_spanned[0]].get_video(
+                start_frame=relative_start, end_frame=relative_stop
+            )
+
+        video_shape = (stop - start,) + self._imaging_extractors[0].get_image_size()
+        video = np.empty(shape=video_shape, dtype=self.get_dtype())
+        current_frame = 0
+
+        # Left endpoint; since more than one extractor is spanned, only care about indexing first start frame
+        relative_start = start - self._start_frames[extractors_spanned[0]]
+        relative_span = self._end_frames[extractors_spanned[0]] - start
+        array_frame_slice = slice(current_frame, relative_span)
+
+        video[array_frame_slice, ...] = self._imaging_extractors[extractors_spanned[0]].get_video(
+            start_frame=relative_start
+        )
+        current_frame += relative_span
+
+        # All inner spans can be written knowing only how long each section is
+        for extractor_index in extractors_spanned[1:-1]:
+            relative_span = self._end_frames[extractor_index] - self._start_frames[extractor_index]
+            array_frame_slice = slice(current_frame, current_frame + relative_span)
+            video[array_frame_slice, ...] = self._imaging_extractors[extractor_index].get_video()
+            current_frame += relative_span
+
+        # Right endpoint; since more than one extractor is spanned, only care about indexing final end frame
+        relative_stop = stop - self._start_frames[extractors_spanned[-1]]
+        array_frame_slice = slice(current_frame, None)
+        video[array_frame_slice, ...] = self._imaging_extractors[extractors_spanned[-1]].get_video(
+            end_frame=relative_stop
+        )
```

Comment:
There is undoubtedly a mathematical expression for including each of these steps into a single iteration loop, but I feel like this is easier to read and comprehend